mlir/lib/IR/MLIRContext.cpp

```diff
@@ -190,6 +190,9 @@ class MLIRContextImpl {
   /// and efficient `getRegisteredOperations` implementation.
   SmallVector<RegisteredOperationName, 0> sortedRegisteredOperations;
 
+  /// This returns the number of registered operations for the given dialect.
+  size_t operationCount = 0;
+
```

Comment:
The comment says that this count is "for the given dialect", but we're in the context class, so it seems to me that you implemented a global count that isn't dialect specific.

The suggestion was to put this counter in `class Dialect {...}` instead I believe

Reply:
I have updated the logic that doesn't require the count now.